data-processing/entities/definitions/model/trainer.py

```diff
@@ -344,325 +284,122 @@ def evaluate_from_input(self, dataset: TensorDataset) -> Tuple[List[int], List[L
-            "Running evaluation on %s dataset, with number of examples = %d, "
-            + " batch size = %d",
-            mode,
-            len(dataset),
-            self.args.eval_batch_size,
-        )
-        eval_loss = 0.0
-        nb_eval_steps = 0
-        intent_preds = None
-        slot_preds = None
-        slot_conf = None
-        gold_intent_label_ids = []
-        gold_slot_labels_ids = []
-
-        self.model.eval()
-
-        for batch in tqdm(eval_dataloader, desc="Evaluating"):
-            batch = tuple(t.to(self.device) for t in batch)
-            with torch.no_grad():
-                inputs = {
-                    "input_ids": batch[0],
-                    "attention_mask": batch[1],
-                    "intent_label_ids": batch[3],
-                    "slot_labels_ids": batch[4],
-                }
-                if self.model.config.model_type != "distilbert":
-                    inputs["token_type_ids"] = batch[2]
-
-                if self.args.use_pos:
-                    inputs["pos_label_ids"] = batch[5]
-                if self.args.use_np:
-                    inputs["np_label_ids"] = batch[6]
-                if self.args.use_vp:
-                    inputs["vp_label_ids"] = batch[7]
-                if self.args.use_entity:
-                    inputs["entity_label_ids"] = batch[8]
-                if self.args.use_acronym:
-                    inputs["acronym_label_ids"] = batch[9]
-
-                outputs = self.model(**inputs)
-                tmp_eval_loss, (intent_logits, slot_logits) = outputs[:2]
+            input_ids_all = inputs["input_ids"].detach().cpu().numpy()
+            # POS
+            pos_ids_all = inputs["pos_label_ids"].detach().cpu().numpy()
 
-                eval_loss += tmp_eval_loss.mean().item()
-
-            nb_eval_steps += 1
-
-            # Intent prediction
-            if intent_preds is None:
-                intent_preds = intent_logits.detach().cpu().numpy()
-                gold_intent_label_ids = (
-                    inputs["intent_label_ids"].detach().cpu().numpy()
-                )
-            else:
-                intent_preds = np.append(
-                    intent_preds, intent_logits.detach().cpu().numpy(), axis=0
-                )
-                gold_intent_label_ids = np.append(
-                    gold_intent_label_ids,
-                    inputs["intent_label_ids"].detach().cpu().numpy(),
-                    axis=0,
-                )
+            # Predict intent
+            intent_probs = {}
+            for data_type, logits in intent_logits.items():
+                intent_probs[data_type] = torch.softmax(logits, dim=1).detach().cpu().numpy()
+                gold_intent_label_dict[data_type] = inputs["intent_label_ids"][data_type].detach().cpu().numpy()
+                intent_preds[data_type] = logits.detach().cpu().numpy()
+                intent_conf[data_type] = intent_probs[data_type][:,1]
```

Comment:
Hi @dykang and @nuwandavek! These two lines are overwriting the `intent_preds` and `intent_conf` arrays with each batch processed, instead of appending to them. The same might also happening for the other dictionaries of: gold_slot_labels_ids, slot_preds, slot_conf, and gold_intent_label_dict.

The result is that if someone passes in a batch of > 16 examples, they will get `NUM_EXAMPLES % 16` predictions. Could one of you take a look at this and offer a fix?